r/R/dataset-write.R

```diff
@@ -206,9 +206,13 @@ write_dataset <- function(dataset,
   validate_positive_int_value(min_rows_per_group)
   validate_positive_int_value(max_rows_per_group)
 
+  source_schema <- source_data(dataset)$schema
+  # For backwards compatibility with Scanner-based writer (arrow <= 7.0.0):
+  # retain metadata from source dataset
```

Comment:
If we had one already, a jira would be nice here, but I'm sure we'll remember this is where it's going even without it, so let'snot